git-keeper-client/gkeepclient/client_configuration.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2016 Nathan Sommer and Ben Coleman
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import configparser
+import os
+
+
+# Store one module-level ClientConfiguration instance
+config_instance = None
+
+
+class ConfigurationError(Exception):
+    pass
+
+
+class ClientConfiguration:
+    """Parses client.cfg and stores the configuration values as attributes.
+
+    Public attributes:
+
+    Server-related attributes:
```

Comment:
I know there are not really "private" data members, but shouldn't we still have getters?  I can't imagine a time we would want to process the data in the config file, but if so, it would be a pain to track down all references.